### DISCUSSION TITLE
Reverse engineering and source code publishing

### DISCUSSION THREAD

**Discussion by rxznve** [community/community#56784] [Programming Help]
Hello!

I am a beginner C/C++ reverse engineer and I really like to develop in this direction. I would like to share my experience with others, but nowhere does it say if it is allowed to publish the source of any cheats for video games and the like on github?

ChatGPT says it's not allowed and I've seen several posts from people complaining that their account was deleted for this reason, however, along with that, I see that there are many popular repositories that contain anti-cheat bypasses and all sorts of other "doubtful" developments.

https://github.com/danielkrupinski/Osiris
https://github.com/AmazingPP/subVerison_GTAV_Hack
https://github.com/cMiraka/Guarded-Regions-Bypass-Valorant

I have nothing against their authors, I just would like to understand, if it is forbidden, then why these repositories are not removed?

**Comment by airtower-luna** [comment]
Do *not* rely on ChatGPT for legal advice, or any other advice for that matter. If you do, the result may be an entertaining read, but very much not a fun situation to be in, like here: https://www.nytimes.com/2023/05/27/nyregion/avianca-airline-lawsuit-chatgpt.html Reverse engineering is a complex legal topic (you might even say legal minefield), but saying it's illegal in principle is simply wrong. You have to look at the specific case, and possibly get *actual* legal advice to be sure.

There might also be an "as long as no-one complains" thing in play. For example, a copyright complaint would need to come from the copyright holder (or their lawyer), not some random person. :wink:

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @rxznve ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by RamtinKosari** [comment]
Hi dear @rxznve 
The legality of publishing cheat software for video games can vary depending on several factors, including the jurisdiction you are in and the specific circumstances surrounding the cheat software. In many cases, creating or distributing cheats for video games may violate the terms of service or end-user license agreements (EULAs) set by the game developers or publishers. This could potentially lead to consequences such as being banned from the game or facing legal action.

Additionally, some countries have laws that prohibit the development or distribution of software that circumvents digital rights management (DRM) or engages in unauthorized access to computer systems, which could include cheats for video games.

To ensure compliance and avoid potential legal issues, it's advisable to:

Review the terms of service and EULAs of the games you are targeting to understand their policies regarding cheats and modifications.
Consider joining communities or forums dedicated to reverse engineering and game hacking, where you can discuss your experiences and share knowledge with others who have similar interests.
Be cautious about publishing or distributing cheat software or related materials, especially if it violates the terms of service or EULAs.
Focus on learning and discussing the technical aspects of reverse engineering and software analysis without specifically targeting cheats for video games.
It's always a good idea to consult with a legal professional who is knowledgeable about intellectual property and software laws in your jurisdiction for specific advice tailored to your situation. They can provide you with accurate information on what is allowed and guide you accordingly.

Remember, it's essential to respect the rights of game developers and publishers while engaging in reverse engineering activities and to use your skills responsibly and ethically.
